optuna/storages/base.py

```diff
@@ -29,158 +28,497 @@ class BaseStorage(object, metaclass=abc.ABCMeta):
+                Name of a new study to create.
 
+        Returns:
+            Study ID, which is an unique id among studies, of the created study.
+
+        Raises:
+            :exc:`optuna.exceptions.DuplicatedStudyError`:
+                If a study with the same ``study_name`` already exists.
+        """
+        # TODO(ytsmiling) Fix RDB storage implementation to ensure unique `study_id`.
         raise NotImplementedError
 
     @abc.abstractmethod
-    def set_study_user_attr(self, study_id, key, value):
-        # type: (int, str, Any) -> None
+    def delete_study(self, study_id: int) -> None:
+        """Delete a study specified by the study ID.
+
+        Args:
+            study_id:
+                Study ID of a study to delete.
 
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching ``study_id`` exists.
+        """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def set_study_direction(self, study_id, direction):
-        # type: (int, study.StudyDirection) -> None
-
+    def set_study_user_attr(self, study_id: int, key: str, value: Any) -> None:
+        """Register a user-defined attribute to a study.
+
+        This method overwrites an existing attribute.
+
+        Args:
+            study_id:
+                Study ID of a study to update.
+            key:
+                Key of an attribute.
+            value:
+                Attributes' value.
+
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching ``study_id`` exists.
+        """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def set_study_system_attr(self, study_id, key, value):
-        # type: (int, str, Any) -> None
+    def set_study_system_attr(self, study_id: int, key: str, value: Any) -> None:
+        """Register an optuna-internal attribute to a study.
+
+        This method overwrites an existing attribute.
+
+        Args:
+            study_id:
+                Study ID of a study to update.
+            key:
+                Key of an attribute.
+            value:
+                Attributes' value.
+
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching ``study_id`` exists.
+        """
+        raise NotImplementedError
 
+    @abc.abstractmethod
+    def set_study_direction(self, study_id: int, direction: study.StudyDirection) -> None:
+        """Register a direction of optimization problem to a study.
+
+        Args:
+            study_id:
+                Study ID of a study to update.
+            direction:
+                Either :obj:`~optuna.study.StudyDirection.MAXIMIZE` or
+                :obj:`~optuna.study.StudyDirection.MINIMIZE`.
+
+        Raises:
+            :exc:`KeyError`:
+                If no study with the matching ``study_id`` exists.
+            :exc:`ValueError`:
+                If `direction` is already set and the passed ``direction`` is the opposite
```

Comment:
Might be missing something but should this not be plain text?
```suggestion
                If direction is already set and the passed ``direction`` is the opposite
```